### ISSUE TITLE
Icon not copied to AppImage

### ISSUE THREAD

**Issue by tresf** [probonopd/linuxdeployqt#145]
A `.png` residing in `<AppDir>/usr/applications/share/pixmaps/<project>.png` is not copied into the `AppImage`, thus resulting in a failed subsequent `-appimage` step.

The desktop file that is being references looks (something) like this:
```properties
[Desktop Entry]
Name=FOO
GenericName=lorem ipsum
GenericName[ca]=lórem ipsum
GenericName[de]=lorem ipzum
GenericName[fr]=la lorem ipsum
Comment=dolor sit amet
Comment[ca]=dólór sit àmet!
Comment[fr]=dolor sit amet !
Icon=foo
Exec=foo.real
Terminal=false
Type=Application
Categories=Qt;AudioVideo;Audio;Midi;
MimeType=application/x-foo-project;
```

Reference  https://github.com/probonopd/lmms/commit/f5fa7450a06d7027e313529f560160e0786fcaf6

**Comment by probonopd** [comment]
`usr/applications/share/pixmaps/<project>.png` is wrong, `usr/share/pixmaps/<project>.png` would be more like it. But the `pixmaps` directory is not the best place to put an icon due to [this](https://github.com/AppImage/AppImageKit/issues/258#issuecomment-254427849).

So better place a 256x256 pixels icon into `usr/share/icons/hicolor/256x256/apps/`.

**Comment by TheAssassin** [comment]
Yes, the standardized path is `/usr/share/icons`. Why is it even in this weird directory? What's the app you're packaging and how is it installed? If it is a third party package, you might want to contact the author because they're obviously doing it wrong.

**Comment by tresf** [comment]
The image is `512px x 512px` and part of a project I help maintain so we'll get it fixed right away.  We'll it into the corresponding folder as directed

The `install` location was decided before my time so that makes sense.  We'll move it to the standardized location and close once confirmed, thanks.

**Comment by probonopd** [comment]
512 pixels is fine too, please place it in `usr/share/icons/hicolor/512x512/apps/`.

**Comment by tresf** [comment]
Per freedesktop.org:

> For example, installing a svg icon in `$prefix/share/icons/hicolor/scalable/apps` means most desktops will have one icon that works for all sizes. 

We have a `foo.svg` as well.  This seems more universal than a specific size.  I assume that's OK?  Then we can abolish the 512x512 which is otherwise redundant.  It's a direct rendering of the SVG.

**Comment by tresf** [comment]
> `usr/applications/share/pixmaps/<project>.png` is wrong

This was a [typo](https://github.com/LMMS/lmms/pull/3688/files#diff-5b07d65c1f1bdd29f324c363c12c9c9bR82) also observable (correctly) [here](https://github.com/probonopd/lmms/commit/f5fa7450a06d7027e313529f560160e0786fcaf6).  The install location is and was `usr/share/pixmaps/<project>.png`.  Does that change the status of this bug?  

freedesktop.org [states](https://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html):

> Icons and themes are looked for in a set of directories. By default, apps should look in `$HOME/.icons` (for backwards compatibility), in `$XDG_DATA_DIRS/icons` and in `/usr/share/pixmaps` (in that order). 

We'd happily adopt the new location (and format) if that's preferred moving forward, but does this change the status of the bug from a `linuxdeployqt` standpoint?

**Comment by tresf** [comment]
Update, the svg residing in `/usr/share/icons/hicolor/scalable/apps/<project>.svg` was copied properly and did not require the workaround.

```
Found icons from desktop file: 
("/home/ubuntu/<project>/build/<project>.AppDir/usr/share/icons/hicolor/scalable/apps/<project>.svg", 
"/home/ubuntu/<project>/build/<project>.AppDir/usr/share/<project>/backgrounds/<project>_tile.png")

iconToBeUsed:
"/home/ubuntu/<project>/build/<project>.AppDir/usr/share/icons/hicolor/scalable/apps/<project>.svg"

Copied "/home/ubuntu/<project>/build/<project>.AppDir/usr/share/icons/hicolor/scalable/apps/<project>.svg"
    to "/home/ubuntu/<project>/build/<project>.AppDir/<project>.svg"
```

**Comment by probonopd** [comment]
`usr/share/pixmaps` means that `appimaged` has a hard time determining the correct icon size which apparently is an issue for some desktops. So if `usr/share/icons/hicolor/scalable/apps` is acceptable, please use that. Still keeping this issue around for reference.

**Comment by tmelorc** [comment]
What about other icons we use for buttons, for example? Where should we put them?

**Comment by probonopd** [comment]
Those can go to `usr/share/pixmaps` or `/usr/share/<APPNAME>`of course, but your application needs to constuct the path to load them relative to the main executable, e.g., using `QString QCoreApplication::applicationFilePath()`.

Reference:
https://github.com/AppImage/AppImageKit/wiki/Bundling-Qt-apps
